layers/image_state.cpp

```diff
@@ -762,3 +769,117 @@ VkSurfaceCapabilitiesKHR SURFACE_STATE::GetCapabilities(VkPhysicalDevice phys_de
     capabilities_[phys_dev] = result;
     return result;
 }
+
+void SURFACE_STATE::SetCompatibleModes(VkPhysicalDevice phys_dev, const VkPresentModeKHR present_mode,
+                                       std::vector<VkPresentModeKHR> &&compatible_modes) {
```

Comment:
This should also use a span. Avoids creating temp std::vector at call site.

```suggestion
void SURFACE_STATE::SetCompatibleModes(VkPhysicalDevice phys_dev, const VkPresentModeKHR present_mode,
                                       layer_data::span<const VkPresentModeKHR> compatible_modes) {
```

Reply:
Done.